### PR TITLE
PMM-5864 Ports page.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**

### CODE REVIEW

**Review by askomorokhov — APPROVED** [review]

**Review by askomorokhov — APPROVED** [review]

File: docs/setting-up/server/network.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Network
+
+## Ports
+
+This is a list of ports used by the various components of PMM.
+
+For PMM to work correctly, your system's firewall should allow traffic on these ports.
+
+| Destination   | Component     | Protocols | Port/Port Range | Dicertion | Description
```
Comment (shoffman-percona) [inline docs/setting-up/server/network.md:9]:
```suggestion
| Destination   | Component     | Protocols | Port/Port Range | Direction | Description
```

File: docs/setting-up/server/network.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+|---------------|---------------|-----------|-----------------|-----------|---------------------
+| PMM Server    | `pmm-server`  | TCP       |   80            | out       | HTTP server, used for web interface
+| PMM Server    | `pmm-server`  | TCP       |  443            | out       | HTTPS server, used for web interface
+| PMM Server    | `pmm-server`  | TCP       | 7771            | both      | gRPC, used for communication between `pmm-agent`, `pmm-admin`
+| PMM Server    | `pmm-server`  | TCP       | 7772            | out       | HTTP1 server, used for older links like `logs.zip`
+| PMM Server    | `pmm-server`  | TCP       | 7773            | both      | Debugging
+| PMM Client    | `pmm-agent`   | TCP       | 7777            | both      | Default `pmm-agent` listen port
```
Comment (shoffman-percona) [inline docs/setting-up/server/network.md:16]: Are these exposed?  I think they're inside the container only right?  where we do gRPC it's over https so 443

File: docs/setting-up/server/network.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+## Ports
+
+This is a list of ports used by the various components of PMM.
+
+For PMM to work correctly, your system's firewall should allow traffic on these ports.
+
+| Destination   | Component     | Protocols | Port/Port Range | Dicertion | Description
+|---------------|---------------|-----------|-----------------|-----------|---------------------
+| PMM Server    | `pmm-server`  | TCP       |   80            | out       | HTTP server, used for web interface
+| PMM Server    | `pmm-server`  | TCP       |  443            | out       | HTTPS server, used for web interface
```
Comment (shoffman-percona) [inline docs/setting-up/server/network.md:12]: this is also how pmm-agent communicates to pmm-server so there's web traffic AND gRPC traffic over HTTPS

File: docs/setting-up/server/network.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+| PMM Server    | `pmm-server`  | TCP       | 7771            | both      | gRPC, used for communication between `pmm-agent`, `pmm-admin`
+| PMM Server    | `pmm-server`  | TCP       | 7772            | out       | HTTP1 server, used for older links like `logs.zip`
+| PMM Server    | `pmm-server`  | TCP       | 7773            | both      | Debugging
+| PMM Client    | `pmm-agent`   | TCP       | 7777            | both      | Default `pmm-agent` listen port
+| PMM Client    | `pmm-agent`   | TCP       | 42000--51999    | in        | Default range for `pmm-agent` connected agents
+| PMM Client    | `vm-agent`    | TCP       | 8428            | both      | Victoria metrics port
+
+> Notes
+>
+> For `pmm-agent`, the default listen port is 7777. This can be changed in the `pmm-agent.yml` configuration file or with the `pmm-agent` flag `--listen-port`. The default range for agents ports can be changed with the flag `--ports-min` and  `--ports-max`, or in the configuration file.
```
Comment (shoffman-percona) [inline docs/setting-up/server/network.md:22]: @rnovikovP is this true?  last I checked you *could* change the port but nothing would work if you did so...we were hard-coded somewhere else...

File: docs/setting-up/server/network.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+| Destination   | Component     | Protocols | Port/Port Range | Dicertion | Description
+|---------------|---------------|-----------|-----------------|-----------|---------------------
+| PMM Server    | `pmm-server`  | TCP       |   80            | out       | HTTP server, used for web interface
+| PMM Server    | `pmm-server`  | TCP       |  443            | out       | HTTPS server, used for web interface
+| PMM Server    | `pmm-server`  | TCP       | 7771            | both      | gRPC, used for communication between `pmm-agent`, `pmm-admin`
+| PMM Server    | `pmm-server`  | TCP       | 7772            | out       | HTTP1 server, used for older links like `logs.zip`
+| PMM Server    | `pmm-server`  | TCP       | 7773            | both      | Debugging
+| PMM Client    | `pmm-agent`   | TCP       | 7777            | both      | Default `pmm-agent` listen port
+| PMM Client    | `pmm-agent`   | TCP       | 42000--51999    | in        | Default range for `pmm-agent` connected agents
+| PMM Client    | `vm-agent`    | TCP       | 8428            | both      | Victoria metrics port
```
Comment (shoffman-percona) [inline docs/setting-up/server/network.md:18]: What talks to this?  as I understand it this is internal to pmm-agent so while it does create a port binding...does anything besides the also local pmm-agent talk to vmagent on this port?

**Review by shoffman-percona — CHANGES_REQUESTED** [review]
I'm probably too late but left some comments/feedback @JiriCtvrtka you might have to look at my comments...

Also I did this [blog](https://www.percona.com/blog/2020/12/01/foiled-by-the-firewall-a-tale-of-transition-from-prometheus-to-victoriametrics/) that has pictures of how the communication flows between client/server both in the new VM push model as well as the classic pull model (where QAN always pushed)

File: docs/setting-up/server/network.md (hunk 4 above)
Comment (JiriCtvrtka) [reply docs/setting-up/server/network.md:22]: @shoffman-percona please check this ticket if this is your case: https://jira.percona.com/browse/PMM-5831. Thank you.

**Comment by JiriCtvrtka** [comment]
@shoffman-percona Please check changes here. It was changed only to ports which are needed to be exposed.

File: docs/setting-up/server/network.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+This is a list of ports used by the various components of PMM.
+
+For PMM to work correctly, your system's firewall should allow TCP traffic on these ports (UDP is not needed).
+
+Ports to expose:
+
+PMM component | TCP port      | Direction     | Description
+--------------|---------------|---------------|------
+`pmm-server`  |   80          | both          | HTTP server, used for gRPC over HTTP and web interface
```
Comment (atymchuk) [inline docs/setting-up/server/network.md:13]:
```suggestion
`pmm-server`  |   80          | both          | HTTP server, used for gRPC over HTTP and web interface (insecure, use with caution)
```

File: docs/setting-up/server/network.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+`pmm-server`  |  443          | both          | HTTPS server, used for gRPC over HTTPS and web interface
```
Comment (atymchuk) [inline docs/setting-up/server/network.md:14]:
```suggestion
`pmm-server`  |  443          | both          | HTTPS server, used for gRPC over HTTPS and web interface (secure, use of SSL certificates is highly encouraged)
```

File: docs/setting-up/server/network.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+Other ports:
+
+PMM component | TCP port      | Direction     | Description
+--------------|---------------|---------------|------
+`pmm-server`  | 7771          | both          | gRPC, used for communication between `pmm-agent`, `pmm-admin`
+`pmm-server`  | 7772          | out           | HTTP1 server, used for older links like `logs.zip`
+`pmm-server`  | 7773          | out           | Debugging
+`pmm-agent`   | 7777          | out           | Default `pmm-agent` listen port
+`vm-agent`    | 8428          | both          | Victoria metrics port
```
Comment (atymchuk) [inline docs/setting-up/server/network.md:24]:
```suggestion
`vm-agent`    | 8428          | both          | VictoriaMetrics port
```

**Comment by rnovikovP** [comment]
Sorry, haven't seen this before. 

It looks like the page will be the part of Setting Up process. if this is the case - I don't think this is the right place to explain what internal ports we use in PMM for components. the person who setting up PMM will not care what's inside on this stage. 

I think, Architecture page is where we can have this information. 

re port "80": arent we removing it slowly from other places? so why keep here?

**Comment by PaulJacobs-percona** [comment]
> Sorry, haven't seen this before.
> 
> It looks like the page will be the part of Setting Up process. if this is the case - I don't think this is the right place to explain what internal ports we use in PMM for components. the person who setting up PMM will not care what's inside on this stage.
> 
> I think, Architecture page is where we can have this information.
> 
> re port "80": arent we removing it slowly from other places? so why keep here?

It's going to be bigger and better! This will be a landing page for network-related setup steps. We currently don't have any and network problems, I believe, are a significant hurdle during the setup process, especially when using containers. While the detailed listing of ports might move somewhere else, I think it's important to have a common setup page (for both server/client) covering some of the generic issues that can prevent a first-time setup from working.

File: docs/setting-up/server/network.md (hunk 4 above)
Comment (rnovikovP) [reply docs/setting-up/server/network.md]: yes, @JiriCtvrtka  this is the issue  we have now

**Comment by JiriCtvrtka** [comment]
So should I move other ports to architecture page? And about port 80 I can remove it from here completly?

**Comment by PaulJacobs-percona** [comment]
> So should I move other ports to architecture page? And about port 80 should I remove it from here completly?

Leave it where it is. Port 80 is a port so it should be mentioned if it is still possible to use it. When port 80 is no longer used by the application, then we will remove it from the table but mention port 80 is now blocked (for security reasons, presumably).

**Pull request merged**